membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -63,14 +62,14 @@ object AttributesFromZuora extends LoggingWithLogstashFields {
 
       if(fromWhere == "Zuora") {
         attributesFromDynamo map { case (_, dynamoAttributes) =>
-          val canSkipCacheUpdate: Boolean = dynamoAndZuoraAgree(dynamoAttributes, attributes, identityId)
           val zuoraAttributesWithAdfree: Option[Attributes] = attributesWithAdFreeFlagFromDynamo(attributes, dynamoAttributes)
 
-          if(!canSkipCacheUpdate) {
+          val updateRequired = dynamoUpdateRequired(dynamoAttributes, attributes, identityId)
+          if(updateRequired) {
             updateCache(identityId, zuoraAttributesWithAdfree, dynamoAttributeService).onFailure {
```

Comment:
I think this means we don't ever write the projected timestamp when we update the dynamo table. I'm looking at this right now.